src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -0,0 +1,980 @@
+    mEventFlag        = static_cast<TimeSyncEventFlag>(eventFlag);
+}
+
+namespace {
+
+class TimeSynchronizationAttrAccess : public AttributeAccessInterface
+{
+public:
+    // register for the TimeSync cluster on all endpoints
+    TimeSynchronizationAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), Id) {}
+
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+private:
+    CHIP_ERROR ReadTrustedTimeSource(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadDefaultNtp(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadTimeZone(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadDSTOffset(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadLocalTime(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+};
+
+TimeSynchronizationAttrAccess gAttrAccess;
+
+CHIP_ERROR TimeSynchronizationAttrAccess::ReadTrustedTimeSource(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    auto tts = TimeSynchronizationServer::Instance().GetTrustedTimeSource();
+    return aEncoder.Encode(tts);
+}
+
+CHIP_ERROR TimeSynchronizationAttrAccess::ReadDefaultNtp(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    char buffer[DefaultNTP::TypeInfo::MaxLength()];
+    MutableCharSpan dntp(buffer);
+    err = TimeSynchronizationServer::Instance().GetDefaultNtp(dntp);
+    if (err == CHIP_NO_ERROR)
+    {
+        err = aEncoder.Encode(CharSpan(buffer, dntp.size()));
+    }
+    else if (err == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
+    {
+        err = aEncoder.EncodeNull();
+    }
+    return err;
+}
+
+CHIP_ERROR TimeSynchronizationAttrAccess::ReadTimeZone(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
+        auto tzList = TimeSynchronizationServer::Instance().GetTimeZone();
+        for (const auto & tzStore : tzList)
+        {
+            ReturnErrorOnFailure(encoder.Encode(tzStore.timeZone));
+        }
+
+        return CHIP_NO_ERROR;
+    });
+
+    return err;
+}
+
+CHIP_ERROR TimeSynchronizationAttrAccess::ReadDSTOffset(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
+        DSTOffset::TypeInfo::Type dstList = TimeSynchronizationServer::Instance().GetDSTOffset();
+        for (const auto & dstOffset : dstList)
+        {
+            ReturnErrorOnFailure(encoder.Encode(dstOffset));
+        }
+
+        return CHIP_NO_ERROR;
+    });
+
+    return err;
+}
+
+CHIP_ERROR TimeSynchronizationAttrAccess::ReadLocalTime(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    DataModel::Nullable<uint64_t> localTime;
+    CHIP_ERROR err = TimeSynchronizationServer::Instance().GetLocalTime(endpoint, localTime);
```

Comment:
If GetLocalTime failed, are we not potentially encoding garbage?  Followup is OK.

Reply:
Tracked in https://github.com/project-chip/connectedhomeip/issues/27575, resolving this comment for now, further followup on the issue.